fearless/Common/QRCoding/QRService.swift

```diff
@@ -0,0 +1,50 @@
+import RobinHood
+
+protocol QRServiceProtocol: AnyObject {
+    @discardableResult
+    func generate(
+        with qrType: QRType,
+        qrSize: CGSize,
+        runIn queue: DispatchQueue,
+        completionBlock: @escaping (Result<UIImage, Error>?) -> Void
+    ) throws -> Operation
+}
+
+final class QRService {
+    let operationFactory: QROperationFactoryProtocol
+    let operationQueue: OperationQueue
+
+    private let encoder: QREncoderProtocol
+
+    public init(
+        operationFactory: QROperationFactoryProtocol,
+        encoder: QREncoderProtocol = QREncoder(),
```

Comment:
I think it will be more beautiful in case you pass concrete encoder implementation there. This constructor takes protocol as parameter, why not ?